reader-writer-lock/src/main/java/com/iluwatar/reader/writer/lock/ReaderWriterLock.java

```diff
@@ -0,0 +1,147 @@
+package com.iluwatar.reader.writer.lock;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class responsible for control the access for reader or writer
+ */
+public class ReaderWriterLock {
+
+  /**
+   * Mutex for reader
+   */
+  private Object r = new Object();
+
+  /**
+   * Global mutex for reader or writer, use to save the holding object
+   */
+  private Set<Object> g = new HashSet<>();
+
+  /**
+   * Current reader count
+   */
+  private int readerCount = 0;
```

Comment:
rename the private variable to `currentReaderCount` and remove the javadoc comment.